File: perceptron.py
import random

def sign(n):
    if n < 0:
        return -1
    return 1


class Perceptron:

    #Constructor
    def __init__(self):
        self.weight = []
        i = 0
        while i < 2:
            self.weight.append(random.uniform(-1, 1))
            i += 1
        self.bias = random.uniform(-1, 1)


    #Getter methods
    def get_weight(self):
        return self.weight

    def get_bias(self):
        return self.bias

    #Setter methods
    def set_weight(self, weight):
        self.weight = weight

    def set_bias(self, bias):
        self.bias = bias

    #Perceptron's methods
    def guess(self, input):
        output = self.bias
        i = 0
        while i < len(input):
            output += self.weight[i] * input[i]
            i += 1
        return sign(output)

    def adjust_perceptron(self):
        pass

percy = Perceptron()

print("Weights: [{:.2f}, {:.2f}] ".format(percy.get_weight()[0], percy.get_weight()[1]))
print("Bias: {:.2f}".format(percy.get_bias()))

inputs = [100, 1]
print("Guess: {:.0f}".format(percy.guess(inputs)))

File: p5 Perceptron/perceptron.js
class Perceptron {
  constructor() {
    this.weights = [];

    for (let i = 0; i < 2; i++) {
      this.weights[i] = random(-1, 1);
    }

    this.bias = random(-1, 1);
  }

  get_weights() {
    return this.weights;
  }

  set_weights(weights) {
    this.weights = weights;
  }

  get_bias() {
    return this.bias;
  }

  set_bias(bias) {
    this.bias = bias;
  }

  guess(inputs_) {
    if (inputs_.length != 2) {
      console.log("Please fix input length");
      return -10;
    }
    let result = this.bias;

    for (let i = 0; i < inputs_.length; i++) {
      result += inputs_[i] * this.weights[i];
    }
    return sign(result);
  }
}
